### PR TITLE
Feature: Add support to DappAddressRelay 

### PR DESCRIPTION
**Pull request by brunomenezes**
### Summary
Code changes to add support to use the `DappAddressRelay` to inform off-chain machine the address of the Dapp contract. More details  on #185.

### CODE REVIEW

File: packages/ui/src/AddressRelayForm.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    useEffect(() => {
+        if (wait.isSuccess) {
+            if (is(Function, onSuccess))
+                onSuccess({ receipt: wait.data, type: "ADDRESS-RELAY" });
+            form.reset();
+            execute.reset();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [wait.isSuccess, onSearchApplications, form, execute]);
```
Comment (nevendyulgerov) [inline packages/ui/src/AddressRelayForm.tsx:86]: I think we can pass the entirety of dependencies here and remove the `eslint-disable` comment:

```
useEffect(() => {
    ...
}, [
    wait.isSuccess,
    onSearchApplications,
    form,
    execute,
    wait.data,
    onSuccess,
]);
```

**Review by nevendyulgerov — COMMENTED** [review]
I left a couple of comments with suggestions.

File: packages/ui/src/AddressRelayForm.tsx (hunk 1 above)
Comment (brunomenezes) [reply packages/ui/src/AddressRelayForm.tsx]: fixed on [f338ee5](https://github.com/cartesi/rollups-explorer/pull/195/commits/f338ee5beeb9fb8ebcccd1285bf1aff95b58d489), good call. I actually added the `wait.data` and `onSuccess`, but I removed the `onSearchApplications`. The clean-up is delegated to the on-success callback.

File: apps/web/src/components/sendTransaction.tsx [hunk 2]
```diff
@@ -173,6 +180,13 @@ const SendTransaction: FC<DepositProps> = ({
                     onSearchTokens={setMultiTokenId}
                     onSuccess={onSuccess}
                 />
+            ) : depositType === "relay" ? (
+                <AddressRelayForm
```
[the remark this answers is not in the document]
Comment (brunomenezes) [reply apps/web/src/components/sendTransaction.tsx:184]: I did an update on [83a3d3d](https://github.com/cartesi/rollups-explorer/pull/195/commits/83a3d3d013ae99e619d539f64e5594aa64351fe1) and for clarity I am using the [contract#method](https://sepolia.etherscan.io/address/0xF5DE34d6BbC0446E2a45719E718efEbaaE179daE#writeContract) name.

**Review by nevendyulgerov — APPROVED** [review]
LGTM!

**Pull request merged**